app/validators/routing/when_validator.py

```diff
@@ -0,0 +1,168 @@
+from app import error_messages
+from app.validators.validator import Validator
+
+
+class WhenValidator(Validator):
+    def __init__(self, when_clause, referenced_id, questionnaire_schema):
+        super(WhenValidator, self).__init__(when_clause)
+        self.when_clause = when_clause
+        self.referenced_id = referenced_id
+        self.questionnaire_schema = questionnaire_schema
+
+    def validate(self):
+        """
+        Validates any answer id in a when clause exists within the schema
+        Will also check that comparison exists
+        """
+        for when in self.when_clause:
```

Comment:
Can we make the var name consistent, so `when_rule` instead of `clause` since we used the word `rule` in the method names?